### DISCUSSION TITLE
Setting CustomAccentColor

### DISCUSSION THREAD

**Discussion by lebhero** [amwx/FluentAvalonia#610] [General]
```
FluentAvaloniaTheme? FluentAvaloniaTheme = Application.Current?.Styles[0] as FluentAvaloniaTheme;
FluentAvaloniaTheme.CustomAccentColor = Colors.Red;
```

Will not change until i call :

`Application.Current!.RequestedThemeVariant `

Ad set the `RequestedThemeVariant` to a value that is different to the current one.

Is this a bug or am i doing something wrong?

**Comment by timunie** [comment]
@lebhero Similar code works for me:

```cs
Application.Current.Styles.OfType<FluentAvaloniaTheme>().First().CustomAccentColor = AccentColor;
```

So I wonder when do you call this? If you do it inside the constructor, maybe this could be the issue as not all is setup. If so, try to wrap it in `Dispatcher.UiThread.Post()` with priority `Loaded`.

**Reply by lebhero** [reply]
Thank you Tim,
I think the value is set but the GUI is not updating probably... The minimize, maximize button foreground colors are also not updating...
I need for example to click on the icons in the NavigationView to update their foreground color like in image below:

[screen.gif](https://s1.gifyu.com/images/SA2N1.gif)


I am not doing it inside the ctor. I call a function to set the color:


```
 public static void SetAppThemeColor(string themeColor)
 {
     var isColor = Color.TryParse(themeColor, out Color c);
     if (!isColor)
     {
         c = DefaultAppThemeColor;
     }
     Dispatcher.UIThread.Post(()=>
             Application.Current.Styles.OfType<FluentAvaloniaTheme>().First().CustomAccentColor = c,
         DispatcherPriority.Loaded
         );
 }
```

**Reply by timunie** [reply]
Some things do update, others not if I saw it correctly. So please make sure you use DynamicResource for Foreground. If all not help, create a very minimal sample to share here for others to take own look.

**Reply by lebhero** [reply]
Thanks Tim, binding the Foreground to DynamicResource works perfectly.
